eclair-core/src/main/scala/fr/acinq/eclair/router/Validation.scala

```diff
@@ -290,7 +327,7 @@ object Validation {
           log.debug("ignoring {} (stale)", u)
           sendDecision(origins, GossipDecision.Stale(u))
           d
-        } else if (pc.getChannelUpdateSameSideAs(u).exists(_.timestamp >= u.timestamp)) {
+        } else if (pc.getChannelUpdateSameSideAs(u).exists(previous => previous.timestamp >= u.timestamp && previous.shortChannelId == u.shortChannelId)) { // NB: we also check the id because there could be a switch alias->real scid
```

Comment:
Actually not really, this isn't where a sleep is useful, it's because in the `onTransition` in `Channel.scala` we don't send the channel update if the timestamp is the same, so we need some kind of sleep. I improved it to use an `eventually` instead of `Thread.sleep`.